packages/wagmi/src/client.ts

```diff
@@ -208,6 +223,54 @@ export class Web3Modal extends Web3ModalScaffold {
         return tx
       },
 
+      writeContract: async (data: WriteContractArgs) => {
+        const chainId = NetworkUtil.caipNetworkIdToNumber(this.getCaipNetwork()?.id)
+
+        const tx = await wagmiWriteContract(wagmiConfig, {
+          chainId,
+          address: data.tokenAddress,
+          abi: data.abi,
+          functionName: data.method,
+          args: [data.receiverAddress, data.tokenAmount]
+        })
+
+        return tx
+      },
+
+      getEnsAddress: async (value: string) => {
+        try {
+          const address = await wagmiGetEnsAddress(this.wagmiConfig, {
+            name: normalize(value),
+            chainId: 1
+          })
+
+          if (address) {
+            return address
+          }
+
+          return false
+        } catch (error) {
+          return false
+        }
+      },
+
+      getEnsAvatar: async (value: string) => {
+        try {
+          const avatar = await wagmiGetEnsAvatar(this.wagmiConfig, {
+            name: normalize(value),
+            chainId: 1
```

Comment:
Is it using `chainId: 1` because ENS is only on Ethereum?

Reply:
Added a check here if `chainId === mainnet.id`. I think we need a extra check on the blockchainAPI to resolve the wallet address from an ENS name. It is now only possible to resolve a name with the wallet address, not otherwise.